pkg/pkgmgr/dpkg.go

```diff
@@ -264,6 +264,14 @@ func (dm *dpkgManager) installUpdates(ctx context.Context, updates unversioned.U
 		installCmd = `sh -c "apt upgrade -y && apt clean -y && apt autoremove"`
 	}
 
+	const heldCmd = `sh -c "apt-mark showhold"`
+	aptHeld := aptUpdated.Dir(resultsPath).Run(llb.Shlex(heldCmd)).Root()
+	heldPackagesBytes, err := buildkit.ExtractFileFromState(ctx, dm.config.Client, &aptHeld, resultsPath)
+	if err != nil {
```

Comment:
looks like you are receiving this error: 'Error: read /tmp/buildkit-mount1189165548/copa-out: is a directory' since resultsPath (copa-out) is a directory. You would need to save the output of the heldCmd to a file first and read that file.

Reply:
@ashnamehrotra I have fixed the error and now I save the output in statusdOutputFile

Reply:
thanks! I think it would be better if we saved it to a new file, since we are using statusdOutputFile to determine whether or not the image is distroless.